### PR TITLE
Add some helpers to reduce boilerplate in controller binaries.

### PR DESCRIPTION
**Pull request by mattmoor**
You can see the reduction in [this PR](https://github.com/knative/serving/pull/3503) taking advantage of these helpers.

### CODE REVIEW

File: controller/controller.go [hunk 1]
```diff
@@ -39,6 +39,10 @@ const (
 	trueString  = "true"
 )
 
+var (
```
Comment (vagababov) [inline controller/controller.go]: This is a const.

File: controller/controller_test.go [hunk 2]
```diff
@@ -625,3 +630,67 @@ func checkStats(t *testing.T, r *FakeStatsReporter, reportCount, lastQueueDepth,
 		t.Errorf("Reconcile success = %v, wanted %v", got, want)
 	}
 }
+
+type FixedInformer struct {
```
Comment (vagababov) [inline controller/controller_test.go]:
```suggestion
type fixedInformer struct {
```
perhaps?

File: controller/controller_test.go [hunk 3]
```diff
@@ -498,7 +500,8 @@ func TestStartAndShutdownWithErroringWork(t *testing.T) {
 
 	var eg errgroup.Group
 	eg.Go(func() error {
-		return impl.Run(1, stopCh)
+		StartAll(stopCh, impl)
```
Comment (vagababov) [inline controller/controller_test.go]: this (and others) can be redone with channels as the new tests, so sleeps can go away.

File: controller/controller_test.go [hunk 4]
```diff
@@ -625,3 +630,67 @@ func checkStats(t *testing.T, r *FakeStatsReporter, reportCount, lastQueueDepth,
...
+		errCh <- StartInformers(stopCh, fi)
+	}()
+
+	select {
+	case err := <-errCh:
+		t.Errorf("Unexpected send on errCh: %v", err)
+	case <-time.After(1 * time.Second):
+		// Wait a brief period to ensure nothing is sent.
+	}
+
```
Comment (vagababov) [inline controller/controller_test.go]: Do we want a test, were sunk goes from false -> true? ⬛️

File: controller/controller.go (hunk 1 above)
Comment (mattmoor) [reply controller/controller.go]: Added a detailed comment, I can see us changing this default at a per-binary level

File: controller/controller_test.go (hunk 3 above)
Comment (mattmoor) [reply controller/controller_test.go]: Done, but it ends up being longer.

File: controller/controller_test.go (hunk 2 above)
Comment (mattmoor) [reply controller/controller_test.go]: Done

File: controller/controller_test.go (hunk 4 above)
Comment (mattmoor) [reply controller/controller_test.go]: Done

File: controller/controller_test.go (hunk 3 above)
Comment (vagababov) [reply controller/controller_test.go]: But no sleeps! :)

**Pull request merged**